python/Ganga/GPIDev/Lib/File/OutputFileManager.py

```diff
@@ -8,38 +8,45 @@
 
 from Ganga.GPIDev.Base.Proxy import isType, stripProxy, getName
 
-"""
-Checks if the output files of a given job(we are interested in the backend) 
-should be postprocessed on the WN, depending on job.backend_output_postprocess dictionary
-"""
-
+from Ganga.Utility.logging import getLogger
+logger = getLogger()
 
 def outputFilePostProcessingOnWN(job, outputFileClassName):
-    backendClassName = getName(job.backend)
-    
-    backend_output_postprocess = stripProxy(job).getBackendOutputPostprocessDict()
-    if backendClassName in backend_output_postprocess:
-        if outputFileClassName in backend_output_postprocess[backendClassName]:
-            if backend_output_postprocess[backendClassName][outputFileClassName] == 'WN':
-                return True
+    """
+    Checks if the output files of a given job(we are interested in the backend) 
+    should be postprocessed on the WN, depending on job.backend_output_postprocess dictionary
+    """
+    return outputFilePostProcessingOnTest(job, outputFileClassName, 'WN')
 
-    return False
 
+def outputFilePostProcessingOnClient(job, outputFileClassName):
+    """
+    Checks if the output files of a given job(we are interested in the backend) 
+    should be postprocessed on the client, depending on job.backend_output_postprocess dictionary
+    """
+    return outputFilePostProcessingOnTest(job, outputFileClassName, 'client')
 
-"""
-Checks if the output files of a given job(we are interested in the backend) 
-should be postprocessed on the client, depending on job.backend_output_postprocess dictionary
-"""
 
+def outputFilePostProxessingOnSubmit(job, outputFileClassName):
+    """
+    Checks if the output files of a given job(we are interested in the backend)
+    should have been postprocessed on job submission, depending on job.backend_output_postprocess dictionary
+    """
+    return outputFilePostProcessingOnTest(job, outputFileClassName, 'submit')
 
-def outputFilePostProcessingOnClient(job, outputFileClassName):
+
+def outputFilePostProcessingOnTest(job, outputFileClassName, when):
```

Comment:
```should be postprocessed 'when'```
OK I can give this a more general name but I was assuming that looking at the code this was fairly clear